### ISSUE TITLE
Billy Tutorial Run Through

### ISSUE THREAD

**Issue by jackzampolin** [cosmos/sdk-application-tutorial#24]
Thanks for running through this with me @okwme! Here are some issues I saw when working  

- [ ] Update tree on first page
- [ ] Explain tendermint on the second page
- [ ] formating on page 3
- [ ] Capitalize `NewnameserviceApp` properly
- [ ] Add note that codec is explained later on page 3
- [ ] keeper.md - sync coin types and figure that out
- [ ] handlers.md - add note that you don't need code from interface
- [ ] wording at the beginning of SetName.md
- [ ] remove notification about NameID and change tutorial
- [ ] Maybe think about using `sdk.Result` in the handler

More to come

**Comment by okwme** [comment]
- [ ] `queriers.md` line 8 missing the word "as" in section "which acts a sub-router"
- [ ] `cli.md` line 176 doesn't actually say where this content should go (`module_client.go`)
- [ ] `rest.md` should give warning to replace the `sdk-application-tutorial` with their own repo like in other places
- [ ] `app-complete.md` minor formatting in `app.go` section (`var app = &nameserviceApp{` starts one indent too many)
- [ ] `app-complete.md` name of app isn't camelcase (`NewnameserviceApp`)
- [ ] `dep.md` where does the makefile go? what is it called?
- [ ] `dep.md` the warning about replacing `sdk-application-tutorial`: is it wrt the toml file? cause i don't see it in there. or just in general for the whole repo? did a search on mine and it doesn't appear anywhere
- [ ] `cmd/nsd/main.go` missing: 
``` 	
        "path/filepath"
	"io/ioutil"
	"strings"

	"github.com/cosmos/cosmos-sdk/x/auth"

	cfg "github.com/tendermint/tendermint/config"
	sdk "github.com/cosmos/cosmos-sdk/types"

```
and includes unnecessarily:
```
	"github.com/tendermint/tendermint/p2p"
```
- [ ] `cmd/nsd/main.go` has different content from final tutorial state from tutorial instructions:
```
//readme says:
	cmd.Flags().String(cli.HomeFlag, DefaultNodeHome, "node's home directory")
	cmd.Flags().BoolP(flagOverwrite, "o", false, "overwrite the genesis.json file")
	cmd.Flags().String(client.FlagChainID, "", "genesis file chain-id, if left blank will be randomly created")
	cmd.Flags().String(flagMoniker, "", "set the validator's moniker")
	cmd.MarkFlagRequired(flagMoniker)
```
```
// repo says:
	cmd.Flags().String(cli.HomeFlag, DefaultNodeHome, "node's home directory")
	cmd.Flags().String(client.FlagChainID, "", "genesis file chain-id, if left blank will be randomly created")
	cmd.Flags().BoolP(flagOverwrite, "o", false, "overwrite the genesis.json file")
```
- [ ] `cmd/nscli/main.go` includes unnecessarily:
```
	authcmd "github.com/cosmos/cosmos-sdk/x/auth/client/cli"
```
- [ ] `x/nameservice/client/rest/rest.go` users `acc` instead of `addr` for `msg := nameservice.NewMsgSetName(req.Name, req.Value, acc)`
- [ ] `x/auth/client/tx.go` error is `x/nameservice/client/cli/tx.go:12:2: imported and not used: "github.com/okwme/cns/vendor/github.com/cosmos/cosmos-sdk/x/auth/client/cli" as authcmd`
- [ ] `build-run.md` says to `You can now start nsd by calling nsd start. You will see logs begin streaming that represent blocks being produced, this will take a couple of seconds.`. how long do i wait? what kind of logs should I see? i'm seeing an error one that says "Couldn't connect to any seeds" is this normal? also afterwards when I run the next command:
```
nscli query account $(nscli keys show billy --address) \
    --indent --chain-id testchain
```
(i used billy instead of jack) 
I get an error:
```
ERROR: failed to prove merkle proof: expected 0 args, got 1
```
even when I try it with explicit address:
```
nscli query account cosmos19tfjch2j2uzz3pc53jwug7ghhx2y66g97xn8ac  --indent --chain-id testchain
```
I get the same merkle proof error...


also I get an error after running:
```
nscli tx nameservice buy-name billy.id 5mycoin \
    --from     $(nscli keys show billy --address) \
    --chain-id testchain
```
I got the error:
```
ERROR: No account with address cosmos19tfjch2j2uzz3pc53jwug7ghhx2y66g97xn8ac was found in the state.
Are you sure there has been a transaction involving it?
```

**Comment by okwme** [comment]
ahaa, i'm thinking i forgot this step:
```
# Add both accounts, with coins to the genesis file
nsd add-genesis-account $(nscli keys show billy --address) 1000mycoin,1000billyCoin
nsd add-genesis-account $(nscli keys show alice --address) 1000mycoin,1000aliceCoin
```
after doing it again with that step i only get the merkle error when querying Alice

**Comment by okwme** [comment]
ok yup, the billy account (jack account) was able to create, edit, read the new domain. but when the alice account tried to buy it i got:
```bash
$ nscli tx nameservice buy-name billy.id 10mycoin \
    --from     $(nscli keys show alice --address) \
    --chain-id testchain
ERROR: No account with address cosmos1zacyaks57uwszh2vm48p8kh7h6sswqf7gpwady was found in the state.
Are you sure there has been a transaction involving it?
```
this was tried after running `rm -r ~/.nsd` and `rm -r ~/.nscli` and following the `build-run` instructions

**Comment by okwme** [comment]
ok it's cause i didn't run `nsd init --chain-id testchain` the second time around 😱
i'll checkmark the "errors" from the first post which were just my fault
otherwise that step works : )

**Comment by jackzampolin** [comment]
closed by #25
